### DISCUSSION TITLE
Submitting TARDIS room schematics

### DISCUSSION THREAD

**Discussion by VariableVixen** [eccentricdevotion/TARDIS#445] [Schematic sharing]
How do (_can_ even) we create, save, and submit TARDIS room schematics for inclusion in the plugin? For instance, the old VAULT room is... not _bad_, but sub-optimal compared to the ones everyone on my server is using, with double chests, _more_ chests, and more space to move about the room. My server's players modify almost every room to some extent, for instance the workshop: more bookshelves, access from all four sides, moving all of the workstations around it, and adding the new villager job site blocks. I could take screenshots from various angles for the submissions, but I don't know how to make them or how to send them to you.

**Comment by eccentricdevotion** [comment]
`/tardisschematic [save|load|paste] [name]`
https://eccentricdevotion.github.io/TARDIS/schematic-commands
Talks about consoles but applies to rooms as well
https://eccentricdevotion.github.io/TARDIS/schematics

I usually start with an empty room or the template room, paste it into a creative world, make my changes, then save the new schematic. To paste, the schematic needs to be in the _user_schematics_ folder.

Happy to accept improvements (with of course the caveat that if I don't like it then it won't make it in lol)
You can attach schematics to a comment.

**Comment by Chew** [comment]
Hmm, it would be cool to have a site for this, like community submitted schematics would be epic, don't know how active it would be but hey, I need something to do, gimme a few days lol

**Comment by eccentricdevotion** [comment]
We used to have one a few years ago (not hosted by me), but it's no longer active.

**Comment by Chew** [comment]
Don't worry, I'll revive it using code I stole from my meme website. So far I've got signing in/up, submitting and viewing schematics, and that's it. lol

**Reply by Mrs-Feathers** [reply]
i am a developer and i absolutely love the tardis plugin. if you need any help getting it set up or even hosting, i would love to throw my hat in the ring. having a schematics sharing site sounds like a super positive thing for the community and something that could surely make the plugin more popular and friendly for everyone being just that more "moddable"

**Comment by roberestarkk** [comment]
To expand on this question a little bit, is there support for making custom versions of existing rooms?
Like if I wanted to make a 2x2x2 Vault with storage all around and the drop-chest in the centre...
Could I:
a) Actually have a non-console room that is bigger than 1x1
b) Have a room with Vault-like functionality without overwriting the existing vault, and just have 'vault' in the name somewhere or something
c) Have the vault (or smelter, or whatever) automation work within the whole room and not just the regular 16x16 chunk
d) Change the height of the 2x2 to be either one or three high, or must it absolutely be a square?

**Comment by eccentricdevotion** [comment]
a) No, rooms can only be 1x1 chunk
b) Not really, vault functionality is tied to growing the "vault" room - that's when its database record is created. You could add a custom schematic for a room and then use the `/tardis update vault` command to set the drop chest location (and the chunk that it scans for chests)
c) No, because of a)
d) Technically schematics can be taller than they are wide e.g. the gravity wells - but they are a special case handled correctly by the plugin (i.e. the ARS grid is updated on multiple levels). There is no functionality for this with custom rooms.

**Comment by roberestarkk** [comment]
Ah okay, so basically I can do large rooms manually, but basically all the functionality of the plugin assumes 1x1 except in special circumstance... Fair enough, cheers!

**Reply by Mrs-Feathers** [reply]
you could do like a double-wide room by making two rooms that just cut off so they attach to the other side perfectly. you just need to grow them in the correct positions and direction
